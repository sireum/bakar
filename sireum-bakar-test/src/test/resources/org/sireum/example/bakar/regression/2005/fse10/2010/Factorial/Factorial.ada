package SimpleMath is
   --# function Fact (N : Natural) return Natural;

   function Factorial (N : Natural) return Natural;
   --# pre Fact(N) <= Natural'Last;
   --# return Fact (N);
end SimpleMath;

package body SimpleMath is
   function Factorial (N : Natural) return Natural
   is
      Result : Natural := 1;
   begin
      for Term in Integer range 1 .. N loop
         Result := Result * Term;
         --# assert 1 <= Term and Term <= N and
         --#        Result = Fact (Term);
         --!# assert Term > 0 and Result = Fact (Term);
      end loop;
      return Result;
   end Factorial;
end SimpleMath;
