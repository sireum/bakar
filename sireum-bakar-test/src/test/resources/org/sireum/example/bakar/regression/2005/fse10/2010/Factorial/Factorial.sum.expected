-------------------------------------------------------------------------------
                          Semantic Analysis Summary                            
                                POGS GPL 2010                                  
            Copyright (C) 2010 Altran Praxis Limited, Bath, U.K.               
-------------------------------------------------------------------------------

Summary of:

Verification Condition files (.vcg)
Simplified Verification Condition files (.siv)
Proof Logs (.plg)
Dead Path Conjecture files (.dpc)
Summary Dead Path files (.sdp)

"status" column keys:
    1st character:
        '-' - No VC
        'S' - No SIV
        'U' - Undischarged
        'E' - Proved by Examiner
        'I' - Proved by Simplifier by Inference
        'X' - Proved by Simplifier by Contradiction
        'P' - Proved by Simplifier using User Defined Proof Rules
        'C' - Proved by Checker
        'R' - Proved by Review
        'F' - VC is False
    2nd character:
        '-' - No DPC
        'S' - No SDP
        'U' - Unchecked
        'D' - Dead path
        'L' - Live path

in the directory:
/Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Factorial

Summary produced: 14-SEP-2010 14:03:00.33

File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Factorial/simplemath/factorial.vcg
function SimpleMath.Factorial

VCs generated 14-SEP-2010 14:02:59

VCs simplified 14-SEP-2010 14:02:59

VCs for function_factorial :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start | rtc check @ 14      | Inference          | No DPC    |   I-   |
| 2   | start | rtc check @ 14      | Inference          | No DPC    |   I-   |
| 3   | start | rtc check @ 15      | Inference          | No DPC    |   I-   |
| 4   | 16    | rtc check @ 15      | Undischarged       | No DPC    |   U-   |
| 5   | start |    assert @ 16      | Undischarged       | No DPC    |   U-   |
| 6   | 16    |    assert @ 16      | Undischarged       | No DPC    |   U-   |
| 7   | start |    assert @ finish  | Undischarged       | No DPC    |   U-   |
| 8   | 16    |    assert @ finish  | Undischarged       | No DPC    |   U-   |
 -----------------------------------------------------------------------------


File /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Factorial/simplemath_/ptf/fact_measure.vcg
function PTF.Fact_Measure

VCs generated 14-SEP-2010 14:02:59

VCs simplified 14-SEP-2010 14:03:00

VCs for function_fact_measure :
 -----------------------------------------------------------------------------
| #   | From  | To                  | Proved By          | Dead Path | Status |
|-----------------------------------------------------------------------------
| 1   | start |    assert @ finish  | Inference          | No DPC    |   I-   |
 -----------------------------------------------------------------------------


===============================================================================
Summary:

The following subprograms have undischarged VCs (excluding those proved false):

   5  /Users/chalin/wkspdirs/36/sireum/sireum-spark-examples/src/org/sireum/profile/spark/examples/fse10/2010/Factorial/simplemath/factorial.vcg

Proof strategies used by subprograms
-------------------------------------------------------------------------
Total subprograms with at least one VC proved by examiner:              0
Total subprograms with at least one VC proved by simplifier:            2
Total subprograms with at least one VC proved by contradiction:         0
Total subprograms with at least one VC proved with user proof rule:     0
Total subprograms with at least one VC proved using checker:            0
Total subprograms with at least one VC discharged by review:            0

Maximum extent of strategies used for fully proved subprograms:
-------------------------------------------------------------------------
Total subprograms with proof completed by examiner:                     0
Total subprograms with proof completed by simplifier:                   1
Total subprograms with proof completed with user defined rules:         0
Total subprograms with proof completed by checker:                      0
Total subprograms with VCs discharged by review:                        0

Overall subprogram summary:
-------------------------------------------------------------------------
Total subprograms fully proved:                                         1
Total subprograms with at least one undischarged VC:                    1  <<<
Total subprograms with at least one false VC:                           0
                                                                    -----
Total subprograms for which VCs have been generated:                    2


ZombieScope Summary:
-------------------------------------------------------------------------
Total subprograms for which DPCs have been generated:                   0
Total number subprograms with dead paths found:                         0
Total number of dead paths found:                                       0


VC summary:
-------------------------------------------------------------------------
Note: U/R denotes where the Simplifier has proved VCs using one or more user-
defined proof rules.

Total VCs by type:                                       
                            -----------Proved By Or Using------------
                     Total  Examiner Simp(U/R)  Checker Review False Undiscgd
Assert or Post:          5      0      1              0      0      0       4
Precondition check:      0      0      0              0      0      0       0
Check statement:         0      0      0              0      0      0       0
Runtime check:           4      0      3              0      0      0       1
Refinement VCs:          0      0      0              0      0      0       0
Inheritance VCs:         0      0      0              0      0      0       0
===============================================================================
Totals:                  9      0      4              0      0      0       5  <<<
% Totals:                       0%    44%             0%     0%     0%     56% <<<
===================== End of Semantic Analysis Summary ========================
